src/app/core/session/session-service/session.service.ts

```diff
@@ -76,21 +69,21 @@ export abstract class SessionService {
    * Get the session status - whether a user is authenticated currently.
    */
   public isLoggedIn(): boolean {
-    return this.getLoginState().getState() === LoginState.LOGGED_IN;
+    return this.loginState.value === LoginState.LOGGED_IN;
   }
 
   /**
    * Get the state of the session.
    */
-  public getLoginState(): StateHandler<LoginState> {
-    return this.loginState;
+  public get loginState(): BehaviorSubject<LoginState> {
```

Comment:
This does not entirely work. The `SyncedSessionService` uses the login state of the `LocalSessionService` and also modifies it in certain scenarios. The current behavior is equivalent to the previous implementation but of course could be changed to a more strict model.

Reply:
I tried to implement it but there are some problems.

1. The subclasses of the `SessionSerivce` can't access private members of the abstract `SessionService`. But need to be able to set the `syncState` and the `loginState`.
2. `.asObservable` removes the the ability to get the current value of the `BehaviorSubject` which is used quite often.

Especially due to 1. I don't see a pretty way which would allow to use the `Observable` interface while still having some shared logic in the `SessionService`.